tests/unit/backend/wmg/api/test_v1.py

```diff
@@ -555,19 +555,23 @@ def test__query_request_with_filter_dims__returns_elided_filter_dims(
                         "label": "dataset_id_0_name",
                         "collection_label": "dataset_id_0_coll_name",
                         "collection_id": "dataset_id_0_coll_id",
-                    },
```

Comment:
⚠️ This should also have `dataset_id_1`. This may just be a side-effect of the test setup, and line 531 may just need to mock the response to have both datasets returned.

Reply:
good catch. I think you are right that the value is just based on the mock on line 531. At this point it feels like I'm mostly testing the mock which def isnt ideal. I'm working on implementing a test as you described above to more throughly test the secondary filters. Do you think its even worth keeping this test?

Reply:
If the new test is asserting both of the following, then yes, I'd say we can safely eliminate this test: 
1) when a secondary dimension has criteria, the remaining filter values for other secondary dimensions are properly restricted (preferably exercising _all_ the other secondary dimensions)
2) when a secondary dimension has criteria, it's own values are not restricted (preferably testing this for all secondary dimensions as well)

It's generally nice to break up different behavioral requirements into separate tests in order to capture the requirements explicitly, but a single test will suffice if it's explaining all the things it's testing.